contributed_definitions/NXsubstance.nxdl.xml

```diff
@@ -0,0 +1,124 @@
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXsubstance" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         A form of matter with a constant, definite chemical composition.
+         
+         Examples can be single chemical elements, chemical compunds, or alloys.
+         For further information, see https://en.wikipedia.org/wiki/Chemical_substance.
+    </doc>
+    <field name="name">
+        <doc>
+             User-defined chemical name of the substance
+        </doc>
+    </field>
+    <field name="molecular_mass" type="NX_FLOAT" units="NX_MOLECULAR_WEIGHT">
+        <doc>
+             Molecular mass of the substance
+        </doc>
+    </field>
+    <field name="cas_number">
+        <doc>
+             Unique numeric CAS REGISTRY number of the sample chemical content
+             For further information, see https://www.cas.org/.
+        </doc>
+    </field>
+    <field name="cas_name">
+        <doc>
+             CAS REGISTRY name of the sample chemical content
+        </doc>
+    </field>
+    <field name="identifier_cas">
+        <doc>
+             CAS REGISTRY URI
+        </doc>
+        <attribute name="type" type="NX_CHAR">
+            <enumeration>
+                <item value="URL"/>
+            </enumeration>
+        </attribute>
+    </field>
+    <group name="cas_image" type="NXnote">
+        <doc>
+             CAS REGISTRY image
+        </doc>
+    </group>
+    <field name="cas_synonyms">
+        <doc>
+             Synonyms in the CAS system.
+        </doc>
+    </field>
+    <field name="identifier_inchi_str">
+        <doc>
+             String InChi identifier.
+             The InChI identifier expresses chemical structures in terms of atomic connectivity,
+             tautomeric state, isotopes, stereochemistry and electronic charge in order to
+             produce a string of machine-readable characters unique to the respective molecule.
+             For further information, see https://iupac.org/who-we-are/divisions/division-details/inchi/.
+        </doc>
+    </field>
+    <field name="identifier_inchi_key">
```

Comment:
Having both the `identifier_inchi_str` field and the `identifier_inchi_key` seems redundant.  If you have one, you should be able to figure out the other.  If so, do we really need to store (or support storing) both?

Reply:
Resolving.  @paulmillar if you have more feedback, please move to another issue